Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -50,23 +52,40 @@ class MessageManager: EngineWebDelegate {
         subscribeToInAppMessageState()
     }
 
+    deinit {
+        unsubscribeFromInAppMessageState()
+        removeEngineWebView()
+    }
+
     func subscribeToInAppMessageState() {
         // Keep a strong reference to the subscriber to prevent deallocation and continue receiving updates
         // Also, since we do not store strong reference of MessageManager anywhere, not keeping a strong reference of
         // subscriber may result in deallocation of MessageManager and hence dismissal of message unexpectedly.
         inAppMessageStoreSubscriber = {
             let subscriber = InAppMessageStoreSubscriber { [self] state in
-                switch state.currentMessageState {
+                let messageState = state.currentMessageState
+                switch messageState {
                 case .displayed:
                     threadUtil.runMain {
                         self.loadModalMessage()
                     }
 
-                case .dismissed:
+                // Dismiss the message when the message is dismissed or initial state
+                // Initial state may only be received when state is reset while a message was being displayed
+                case .dismissed, .initial:
```

Comment:
This looks vulnerable, specially `.inital` part, when we trigger `loadMessage` action, that's when MessageManager is created and probably this listener is going to be setup. The state at that point is still `.initial`. 

It isn't till `loadMessage` goes to the reducer, then its state is changed from the initial.

Reply:
Discussed this on the call. The `initial` event is not received, and only changes are. So this won't be an issue with current implementation.